packages/application-generic/src/services/filters/filters.service.ts

```diff
@@ -0,0 +1,157 @@
+import * as _ from 'lodash';
+import { StepFilter, SubscriberEntity, TenantEntity } from '@novu/dal';
+import {
+  FieldOperatorEnum,
+  FILTER_TO_LABEL,
+  IBaseFieldFilterPart,
+  ICondition,
+  ITenantDefine,
+  ITriggerPayload,
+} from '@novu/shared';
+
+export interface IFilterData {
+  subscriber?: SubscriberEntity;
+  payload?: any;
+}
```

Comment:
Isn't this one a duplication of IFilterVariables?

Reply:
You are right. It is just a subset. I will try to check if I can reuse IFilterVariables there. Good spot.

Reply:
It works but now I should merge the `data` parameter and `variables` and do it in a way that the different use cases work properly.